spec/term/1.0.md

```diff
@@ -0,0 +1,204 @@
+
+*Rationale:* This rule allows terminals to be backwards-compatible with legacy control sequences, such as those defined in VT100 aka ECMA-48 aka ANSI X3.64 aka ISO/IEC 6429.
+
+Characters that are in the general category Co ("Other, private use") MAY be exempted from the treatment described above if the characters in question have a graphic representation in the font that the terminal uses to display its character array.
+
+Finally, the server MUST discard each U+000A (new line) character that immediately succeeds a U+000D (carriage return) character, and then convert each U+000D character into a U+000A character.
+
+*Rationale:* This rule ensures consistent line break handling across different platforms, i.e., CR+NL = CR = NL.
+
+The remaining string of Unicode characters SHALL then be inserted into the terminal document at the position pointed to by the cursor.
+The cursor SHALL be updated to point at the position directly after the inserted string.
+If the terminal document would grow too large to fit within the program's memory because of this operation, the server MAY choose to discard some of the text at the start of the document.
+If it does so, it SHOULD discard entire lines or paragraphs, that is, the last codepoint discarded SHOULD be U+000A, U+000C, U+2028 or U+2029.
+
+### 1.3. Standard input
+
+TODO describe how keypresses are reflected on stdin (canonical/raw mode, noecho property, etc.), how Ctrl-D is handled
+
+### 1.4. Display
+
+If the server displays the document to a user, or renders it to some graphics device or file, then the terminal document SHALL be laid out according to the rules for normalization, writing direction, line breaking and handling of bidirectional text as defined in the Unicode Standard, as published by the [Unicode consortium](https://www.unicode.org).
+
+The surface onto which the terminal document is rendered is called the **terminal viewport**.
+
+### 1.5. Scrollback
+
+The server MAY impose limits on the size of the terminal document (e.g. because of memory constraints).
+When new text is added to the terminal document in such a way that the resulting terminal document would exceed those limits, the server MAY discard a prefix of the terminal document.
+If the cursor points into that prefix or at its start, the cursor SHALL be moved to the end of the prefix before the prefix is discarded.
+
+The server SHOULD choose the prefix such that its end is a line separator, paragraph separator, or a line break control character (U+0010).
+
+*Rationale:* In other words, servers should prefer to discard whole lines at a time.
+Chomping off only parts of a line causes unnecessary confusion on the side of the user.
+
+## 2. Message types for `vt6/term1`
+
+### 2.1. The `term.make-stdio` message
```

Comment:
`make-stdio` and `is-stdio` should not be part of `term`, but of `core`, since muxers and proxies must also be able to turn a connection into stdio.

Reply:
See above.

Reply:
Let me rephrase: `make-stdio` and `is-stdio` should not be part of `term`, but of `core`, since those are functionalities that *every* server must support, regardless of whether it `have`s `term` or not.
I know, all servers will `have term`, but the point is that the `make-stdio` and `is-stdio` are independent of that.

We must not define things that every server must have inside a non-`core` module that every server must have, when those things don't have anything to do with it semantically!

Reply:
Agreed. Move this to `core` to enforce a logical separation between `core` (which corresponds roughly to OSI layers 1-4) and `term` (which corresponds roughly to OSI layers 5-7).